### ISSUE TITLE
Product design for Async Migrations page

### ISSUE THREAD

**Issue by tiina303** [PostHog/posthog#8100]
Current partial design in https://github.com/PostHog/posthog/pull/7885 

<img width="1765" alt="Screen Shot 2022-01-17 at 16 24 30" src="https://user-images.githubusercontent.com/890921/149807902-87c698ba-872c-44ad-9496-0ebe97038b74.png">

Users will have N migrations (currently 3, assume not added too frequently, in the future we might want to hide older completed ones) and there will be up to 2 they can take an action on the currently running one or the previous one that completed and next one that can run.

Current states and actions:
0. Not started: only one action to trigger the migration start
1. Starting: non-clickable when someone recently clicked start from above we show a spinning wheel until the migration gets to running state.
2. running: one can stop or stop&rollback
3. completed: non-clickable shows a checkmark currently
4. errored: one can resume (continue from whatever progress point we were left of), restart or rollback
5. rolled back: one can restart

Currently we have the error column, but it will actually be a separate table in the future to be able to show all errors tied to current migration, so thinking we can just have that below as a separate table.

The page is used only by few users per self-hosted instance, so not necessarily the highest priority to get the design nailed down and would appreciate any quick suggestions too.

**Comment by mariusandra** [comment]
Looping in @clarkus for design input. Any thoughts on how to best present the "current states and actions" column, and for the rest of the table?

**Comment by clarkus** [comment]
I redrew the table using some of our recently updated table patterns. I've stacked the migration name and  description to save a bit of space. Are there any other actions applicable to each migration status? I've also added in inline copy actions for the ID attributes. 

I wasn't quite sure what to do with the error column. That seems like it should only be applicable when the migration has encountered an error. Maybe we should display the error details in some secondary way?

![Migrations](https://user-images.githubusercontent.com/254612/150012860-8338fd63-5e5f-462f-8b93-62292d49cd1c.png)

![Menu options](https://user-images.githubusercontent.com/254612/150012331-39dd540c-e72f-4533-87ae-c4cf315cc0eb.png)

https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=6616%3A44326

**Comment by tiina303** [comment]
> I wasn't quite sure what to do with the error column. That seems like it should only be applicable when the migration has encountered an error. Maybe we should display the error details in some secondary way?

Mentioned description
> Currently we have the error column, but it will actually be a separate table in the future to be able to show all errors tied to current migration, so thinking we can just have that below as a separate table.

**Comment by tiina303** [comment]
Elaboration on actions what they are and what they do:
1. "start" (for a new migration starts it from the beginning)  - safe
2. "resume" (for a migration that was stopped tries to continue) - warning
3. "restart" a rolled back migration - safe
4. "restart" a stopped migration - ignores that we already made some steps and starts from the beginning as if we hadn't - dangerous
5. "stop and rollback" - dangerous
6. "stop without rollback" - dangerous
7. "rollback" - warning (of completed migration = dangerous)

**Comment by clarkus** [comment]
> "resume" (for a migration that was stopped tries to continue) - warning

What would cause the user to consider not doing this? Is it dangerous in some way or do we need to prompt the user about specific concerns?

> "restart" a stopped migration - ignores that we already made some steps and starts from the beginning as if > we hadn't - dangerous
> "stop and rollback" - dangerous
> "stop without rollback" - dangerous
> "rollback" - warning (of completed migration = dangerous)

For dangerous actions, should we prompt to ensure the user wants to perform the action? Is it an irrecoverable action? If we prompt to confirm the action, we will want to quantify why the action is dangers - will data be lost, can I undo the action, etc?

**Comment by tiina303** [comment]
re dangerous: I think just making it visually a bit more scary looking is good enough 😉 

If we are super awesome at writing migrations this would be all safe and reversible, however since this is new some actions are more likely going to cause problems compared to others.

**Comment by tiina303** [comment]
It's not necessary, we can also just explain that in the doc etc too, so feel free to ignore if that doesn't fit into the design well.

**Comment by clarkus** [comment]
Updated concept at https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=6625%3A32152

Depending on the state of each migration, we'll show an overflow menu or a single action. To align migration states where actions can overlap, I've included icons in the overflow menus as well. Some of these are custom, so I can export them for inclusion in our svg set. 

![Migrations](https://user-images.githubusercontent.com/254612/150044633-d54c34f7-eaf6-4568-9705-ca1ab1a667b5.png)
